---
title: "2c_S8S"
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

```{r}
library(dplyr)
library(devtools)
library(tidyverse)
library(data.table)
library(here)
library(readxl)
library(ggpubr)
library(cowplot)


```

```{r MCCC1, echo=FALSE, fig.width=6, fig.height=6, dev=c("png", "pdf")}

meta<-fread(here("data","focal_pcawg","pcawg_sample_sheet.tsv"))
meta<-meta %>% filter(dcc_project_code %in% c("BRCA-US"))
#cnv<-fread(here("data","focal_pcawg","all_samples.consensus_CN.by_gene.170214.txt"))
#inter<-intersect(meta$aliquot_id,colnames(cnv))
#cnv2<-cnv %>% select(`Gene Symbol`, `Locus ID`, Cytoband, inter)

#cnv2<-cnv2 %>% gather(key=Sample,value=CNV,-`Gene Symbol`,-`Locus ID`,-Cytoband)
#cnv2<-inner_join(cnv2,meta %>% select(icgc_donor_id,dcc_project_code,dcc_specimen_type,aliquot_id),by=c("Sample"="aliquot_id"))
#cnv2<-cnv2 %>% filter(CNV != "NaN")


#erbb2<-cnv2 %>% filter(`Gene Symbol`=="ERBB2")
#write.csv(erbb2,here("data","focal_pcawg","all_samples.consensus_CN.ERBB2.csv"))
#for the purpose of github file size limit, made this filtered version for plotting
erbb2<-fread(here("data","focal_pcawg","all_samples.consensus_CN.ERBB2.csv")) %>% select(-V1)

data<-fread(here("data","Kmer_Distributions","chr_distribution.csv"))
data<-data %>% select(-V1)
data$total_kmers<-data$occurrences*data$kmer_count
data<-data %>% group_by(feature)%>%summarize(total_kmers=sum(total_kmers))

data2<-fread(here("data","Kmer_Distributions","chr_distribution_altemose.csv"))
data2<-data2 %>% select(-V1)
data2$total_kmers<-data2$occurrences*data2$kmer_count
data2<-data2 %>% group_by(feature)%>%summarize(total_kmers=sum(total_kmers))
data<-rbind(data,data2)
#data$total_kmers<-data$total_kmers/(30*.86) #86% of 30 million reads

gene<-fread(here("data","focal_pcawg","ERBB2.txt"))
colnames(gene)<-c("feature","count","ref")

data<-inner_join(data,gene,by="feature")
data$perc<-(data$count/data$total_kmers)*100

data<-data %>% filter(perc != 0)
data<-data %>% filter(perc>.5)

###Check Novelty
novel<-read_csv(here("data","NoveltyTables","Novel_Elements.csv"))
novel %>% filter(feature %in% data$feature) %>% select(feature,Novelty,group) %>% rename("Subfamily status"="Novelty")
###

all_samples<-fread(here("data","TCGA","repository_1680116108.tsv"))
ploidy<-readRDS(here("data","TCGA","tcga_data_export_for_artemis_20230404.rds"))
ploidy<-inner_join(ploidy,all_samples,by=c("dcc_file_id"="File ID"))
exc<-read_xlsx(here("data","TCGA","pcawg-exclusion-list.xlsx"))
ploidy<-ploidy %>% filter(!`ICGC Donor` %in% exc$icgc_donor_id)


pcawg2<-fread(here("data","PCAWG","artemis_PCAWG1.csv"))
pcawg2<-pcawg2 %>% tibble() %>% select(-V1)
#pcawg<-pcawg %>% tibble()
#pcawg2<-pcawg2 %>% filter(!id %in% pcawg$id)
#pcawg<-rbind(pcawg,pcawg2)
pcawg<-pcawg2
pcawg$type<-sapply(str_split(pcawg$id,"_"),"[",2)

pcawg<-pcawg %>% mutate(type=if_else(grepl("tumor",id),"tumor","normal"))
pcawg$id<-sapply(str_split(pcawg$id,"_"),"[",1)
pcawg<-pcawg %>% filter(id %in% ploidy$`ICGC Donor`)

pcawg<-inner_join(pcawg,ploidy %>% select(`ICGC Donor`,project_id),by=c("id"="ICGC Donor"))

pcawg<-pcawg %>% filter(id %in% erbb2$icgc_donor_id)
pcawg<-pcawg %>% select(id,type,data$feature)
pcawg<-tibble(pcawg)
pcawg<-pcawg %>% gather(key=feature,value=count,-id,-type) %>% spread(key=type,value=count)
pcawg<-inner_join(pcawg,erbb2 %>% select(CNV,icgc_donor_id,dcc_project_code),by=c("id"="icgc_donor_id"))
pcawg$ratio<-pcawg$tumor/pcawg$normal

#Definition based on COSMIC gain >= 5 https://cancer.sanger.ac.uk/cosmic/help/cnv/overview
pcawg<-pcawg %>% mutate(focal=if_else(CNV>4,"Focal Gain","CN < 5"))
pcawg<-inner_join(pcawg,data,by="feature")
pcawg$focal<-factor(pcawg$focal,levels=c("CN < 5","Focal Gain"))


pcawg$name<-paste0(pcawg$feature,"\n(",round(pcawg$perc,2),"% of defining kmers)")
pcawg$name2<-paste0(round(pcawg$perc,2),"% of kmers overlap")

f<-(pcawg %>% filter(id=="DO1249")%>%arrange(-perc))$name
pcawg$name<-factor(pcawg$name,levels=f)

```

```{r}
pcawg$feature<-gsub("#","\n",pcawg$feature)
f2<-(pcawg %>% filter(id=="DO1249")%>%arrange(-perc))$feature

pcawg$feature<-factor(pcawg$feature,levels=f2)

pcawg %>% group_by(feature,focal) %>% summarize(m=mean(tumor)) %>% print(n=1000)
wilcox.test((pcawg %>% filter(feature=="HERVIP10F-int\nLTR_ERV1" & focal=="CN < 5"))$tumor,(pcawg %>% filter(feature=="HERVIP10F-int\nLTR_ERV1" & focal=="Focal Gain"))$tumor,alternative="less")
names<-pcawg %>% filter(id=="DO1249")

novel<-fread(here("data","NoveltyTables","Novel_Elements.csv"),header=T)
novel<-novel %>% filter(group=="Novel")
novel$feature<-gsub("#","\n",novel$feature)

pcawg<-pcawg %>% mutate(Novelty=if_else(feature %in% novel$feature,"Newly identified as altered in cancer","Prior Evidence"))


a2<-ggplot(pcawg,aes(x=focal,y=ratio,fill=focal))+geom_boxplot(outlier.shape=NA,width=.5)+
  geom_jitter(size=.2)+facet_grid(feature~.,switch="both")+
  coord_flip()+theme_classic(base_size=14)+
  theme(legend.position="bottom",strip.text.y.left=element_text(angle=0,face="bold"),strip.background = element_blank(),
        axis.text.y=element_blank(),axis.title.y=element_blank(),legend.title=element_blank(),strip.placement="outside",legend.text = element_text(size=18),plot.title = element_text(hjust = 1))+ylab("Ratio of Tumor:Normal kmer count")+labs(fill="")+geom_text(data=names, aes(x = "CN < 5", y = 1.8,label=name2,group=feature),alpha=1)+ggtitle("Breast Cancer - ERBB2 Region")

l2<-get_legend(a2)

a2<-a2 + theme(legend.position="none")

#save_data<-pcawg
#write.csv(save_data %>% select(focal,ratio,feature,id),here("output","Figure_Data_Files","Fig2c_left.csv"))

```


```{r}
p<-compare_means(ratio~focal, pcawg, method = "wilcox.test",p.adjust.method="BH",paired = FALSE,
  group.by = "feature", ref.group = NULL,alternative="greater")
p$feature<-factor(p$feature,levels=f2)
p$logp<--1*log10(p$p.adj)

p<-p %>% mutate(color=if_else(p.adj<=.05,"Significant","Not Significant"))
a4<-ggplot(p,aes(x=feature,y=logp,color=color))+facet_grid(feature~.,scales="free_y")+coord_flip()+geom_point()+theme_classic(base_size=14)+theme(axis.text.y=element_blank(),strip.text=element_blank(),axis.title.y = element_blank(),legend.title=element_blank(),legend.position="bottom",legend.text = element_text(size=18),legend.key.size = unit(3,"line"))+ylab("p-val\n(-log10)")+scale_color_manual(values=c("gray50","indianred"))+geom_hline(yintercept=-1*log10(.05),linetype="dashed",alpha=.5)+guides(color=guide_legend(override.aes = list(size = 4)))

#-1*log10(.05/13)

l<-get_legend(a4)

#save_data<-p
#write.csv(save_data %>% select(feature,logp,color),here("output","Figure_Data_Files","Fig2c_right.csv"))

```



```{r comp2, echo=FALSE, fig.width=8, fig.height=12, dev=c("png", "pdf")}
a4<-a4 + theme(legend.position="none")

p1<-plot_grid(a2,a4,align="h",axis="tb",ncol=2,rel_widths=c(3,.7))
p2<-plot_grid(l2,l,align="h",axis="b",ncol=2,rel_widths=c(1,1))
p1
#p3<-plot_grid(p1,p2,nrow=2,rel_heights=c(10,1))
#p3
```

```{r comp_legend, echo=FALSE, fig.width=10, fig.height=1.5, dev=c("png", "pdf")}
p2
#p3<-plot_grid(p1,p2,nrow=2,rel_heights=c(10,1))
#p3
```



```{r p2, echo=FALSE, fig.width=12, fig.height=3, dev=c("png", "pdf")}

cn<-pcawg %>% filter(feature=="HERVIP10F-int\nLTR_ERV1")
cn$feature<-"Spare for CNV"
cn$set<-"ERBB2 Copy Number"

c<-ggplot(cn,aes(x=CNV,fill=focal))+geom_histogram(binwidth=1)+
  theme_classic(base_size=14)+
  theme(legend.position="bottom")+xlab("Tumor ERBB2 Copy Number")+ylab("Count")
c


```


